Shared/Supporting Files/Views/CategoriesView.swift

```diff
@@ -49,12 +49,29 @@ struct CategoryGridView: View {
                     .buttonStyle(.plain)
                 }
             }
-            .padding(8)
+            .padding()
         }
     }
 }
 
-private extension CategoryGridView {
+private extension CategoriesView {
+    struct CategoryList: View {
+        /// The samples in a category.
+        let samples: [Sample]
+        
+        var body: some View {
+            List(samples, id: \.name) { sample in
+                NavigationLink {
+                    SampleDetailView(sample: sample)
+                        .id(sample.name)
+                } label: {
+                    SampleRow(sample: sample, query: "")
```

Comment:
I couldn't use the synthesized initializer for some reason. Maybe I need to look up the Swift doc 🤔 , but I vaguely recall it retains the default initializer even if a "convenience" one is added to a struct, and per https://www.swiftbysundell.com/tips/struct-convenience-initializers/

Reply:
You'll notice in the code in the linked article, they are adding an initializer in an extension. Adding custom initializers in the struct declaration itself will suppress generation of the memberwise initializer.